pinot-core/src/main/java/org/apache/pinot/core/query/request/context/QueryContext.java

```diff
@@ -0,0 +1,199 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.query.request.context;
+
+import java.util.List;
+import java.util.Map;
+import org.apache.pinot.common.request.BrokerRequest;
+
+
+/**
+ * The {@code QueryContext} class encapsulates all the query related information extracted from the wiring Object.
```

Comment:
Is it correct to say that that this can be constructed on the server regardless of whether the wire object is BrokeRequest or PinotQuery. So we need to have 2 converters to be used upon request deserialization on the server side?

- BrokerRequest to QueryRequest
- PinotQuery to QueryRequest

Reply:
Yes and no. We always use `BrokerRequest` as the wire object right now, and it has the `PinotQuery` inside for SQL queries. We only need one converter from BrokerRequest to QueryRequest, but when PinotQuery exists, it will try to convert from PinotQuery.

When we switch the wire object, all we need to do is add a new converter without any change on the query engine